apps/web/src/pages/Pools/EditPoolPage/components/ScreeningQuestions.tsx

```diff
@@ -0,0 +1,356 @@
+import {
+  Repeater,
+  TextArea,
+  Submit,
+  WordCounter,
+  countNumberOfWords,
+} from "@gc-digital-talent/forms";
+import { notEmpty } from "@gc-digital-talent/helpers";
+
+import {
+  CreateScreeningQuestionInput,
+  UpdateScreeningQuestionInput,
+  LocalizedString,
+  PoolAdvertisement,
+  Scalars,
+  AdvertisementStatus,
+  UpdatePoolAdvertisementInput,
+} from "~/api/generated";
+import { EditPoolSectionMetadata } from "~/types/pool";
+
+import { useEditPoolContext } from "./EditPoolContext";
+
+const TEXT_AREA_ROWS = 3;
+const TEXT_AREA_MAX_WORDS = 200;
+
+type ScreeningQuestionValue = {
+  id?: Scalars["ID"];
+  question: LocalizedString;
+};
+
+type FormValues = {
+  questions?: Array<ScreeningQuestionValue>;
+};
+
+export type ScreeningQuestionsSubmitData = Pick<
+  UpdatePoolAdvertisementInput,
+  "screeningQuestions"
+>;
+
+const WordLimit = ({ locale, index }: { locale: Locales; index: number }) => {
+  const value = useWatch({ name: "questions" });
+
+  return (
+    <div data-h2-margin="base(-x.5, 0, x1, 0)" data-h2-text-align="base(right)">
+      <WordCounter
+        text={value[index].question[locale] || ""}
+        wordLimit={TEXT_AREA_MAX_WORDS}
+      />
+    </div>
+  );
+};
+
+const maxWordValidator = (value: string, intl: IntlShape) => {
+  return (
+    countNumberOfWords(value) <= TEXT_AREA_MAX_WORDS ||
+    intl.formatMessage(errorMessages.overWordLimit, {
+      value: TEXT_AREA_MAX_WORDS,
+    })
+  );
+};
+
+interface ScreeningQuestionsProps {
+  poolAdvertisement: PoolAdvertisement;
+  sectionMetadata: EditPoolSectionMetadata;
+  onSave: (submitData: ScreeningQuestionsSubmitData) => void;
+}
+
+const ScreeningQuestions = ({
+  poolAdvertisement,
+  sectionMetadata,
+  onSave,
+}: ScreeningQuestionsProps) => {
+  const intl = useIntl();
+  const [removed, setRemoved] = React.useState<Array<string>>([]);
+  const { isSubmitting } = useEditPoolContext();
+
+  const dataToFormValues = (initialData: PoolAdvertisement): FormValues => ({
+    questions:
+      initialData?.screeningQuestions
+        ?.filter(notEmpty)
+        .map(({ id, question }) => ({
+          id: id || "new",
+          question: {
+            en: question?.en || "",
+            fr: question?.fr || "",
+          },
+        })) || [],
+  });
+
+  const methods = useForm<FormValues>({
+    defaultValues: dataToFormValues(poolAdvertisement),
+  });
+  const { handleSubmit, control } = methods;
+  const { remove, move, append, fields } = useFieldArray({
+    control,
+    name: "questions",
+  });
+
+  const handleRemove = (index: number) => {
+    const toBeRemoved = fields[index];
+    setRemoved((currentlyRemoved) => {
+      if (!currentlyRemoved.includes(toBeRemoved.id)) {
+        return [...currentlyRemoved, toBeRemoved.id];
+      }
+
+      return currentlyRemoved;
+    });
+
+    remove(index);
+  };
+
+  const handleSave = (formValues: FormValues) => {
+    const create: Array<CreateScreeningQuestionInput> = [];
```

Comment:
Oof, we should have added a sync field to the API to avoid this type of diffing.